soroban-env-host/src/host.rs

```diff
@@ -889,6 +912,13 @@ impl Host {
         args: &[RawVal],
         allow_reentry: bool,
     ) -> Result<RawVal, HostError> {
+        if self.is_debug() {
+            // This event will not be rolled back if the call to func fails
```

Comment:
After thinking about this some more, we should rollback the event that says `func` was called if it failed. This could be inferred if the return event wasn't seen, but that's a subtle detail that can be missed.